src/core/execute.c

```diff
@@ -3169,6 +3169,7 @@ static int apply_mount_namespace(
                         .protect_system = context->protect_system,
                         .protect_proc = context->protect_proc,
                         .proc_subset = context->proc_subset,
+                        .mount_nosuid = context->no_new_privileges,
```

Comment:
maybe add a quick comment here, because this is not a 1:1 mapping unike the earlier lines, but comes with some thining behind it. i.e. a brief comment like this: `/* If NNP is on, we can turn on MS_NOUID, since it won't have any effect anymore. */`